tfx/extensions/experimental/kfp_compatibility/kfp_container_component.py

```diff
@@ -0,0 +1,112 @@
+
+  Example:
+    component = load_kfp_yaml_container_component(
+      "kfp_pipelines_root/components/datasets/Chicago_Taxi_Trips/component.yaml"
+    )
+
+  Args:
+    path: local file path of a Kubeflow Pipelines component
+                         YAML file.
+
+  Returns:
+    Container component that can be instantiated in a TFX pipeline.
+  """
+  with open(path) as component_file:
+    data = yaml.load(component_file, Loader=yaml.FullLoader)
+  convert_target_fields_to_kv_pair(data)
+  component_spec = json_format.ParseDict(
+      data, kfp_component_spec_pb2.ComponentSpec())
+  container = component_spec.implementation.container
+  command = (list(map(get_command_line_argument_type, container.command)) +
+             list(map(get_command_line_argument_type, container.args)))
+  # TODO(ericlege): Support classname to class translation in inputs.type
+  inputs = {item.name: File for item in component_spec.inputs}
+  outputs = {item.name: File for item in component_spec.outputs}
+  parameters = {}
+  return container_component.create_container_component(
+      name=component_spec.name,
+      image=container.image,
+      command=command,
+      inputs=inputs,
+      outputs=outputs,
+      parameters=parameters,
+    )
+
+
+def convert_target_fields_to_kv_pair(
+    parsed_dict: Dict[Text, Any]
+) -> None:
+  """ Converts in place specific string fields to key value pairs of {constantValue: [Text]} for proto3 compatibility.
+
+  Args:
+    parsed_dict: dictionary obtained from parsing a Kubeflow component spec.
+                 This argument is modified in place.
+
+  Returns:
+    None
+  """
+  conversion_string_paths = [
+      ['implementation', 'container', 'command'],
+      ['implementation', 'container', 'args'],
+  ]
+  for path in conversion_string_paths:
+    parsed_dict_location = parsed_dict
+    for label in path:
+      parsed_dict_location = parsed_dict_location.get(label, {})
+    if isinstance(parsed_dict_location, list):
+      for ind, value in enumerate(parsed_dict_location):
+        if isinstance(value, str):
+          parsed_dict_location[ind] = {"constantValue": value}
+
+
+def get_command_line_argument_type(
+    command: kfp_component_spec_pb2.StringOrPlaceholderer
+) -> executor_specs.CommandlineArgumentType:
+  """ Converts a container command to the corresponding type under executor_specs.CommandlineArgumentType.
+
+  Args:
+    command: StringOrPlaceholderer which encodes a container command.
+
+  Returns:
+    command to be passed into create_container_component.
+  """
+  if command.HasField("constantValue"):
+    return command.constantValue
+  if command.HasField("inputValue"):
+    return placeholders.InputValuePlaceholder(command.inputValue)
+  if command.HasField("inputPath"):
+    return placeholders.InputUriPlaceholder(command.inputPath)
+  if command.HasField("outputPath"):
+    return placeholders.OutputUriPlaceholder(command.outputPath)
```

Comment:
I'm not sure `OutputUriPlaceholder` can work as substitute for the `OutputPathPlaceholder`. See my comment in the PR.

Reply:
You're right that this does not capture the same semantics yet.  CC'd you on an internal discussion for the downloading semantics for possible ways forward.